presto/eval/eval.py

```diff
@@ -1,25 +1,46 @@
+import logging
+from abc import ABC
+from dataclasses import astuple, dataclass
 from typing import Callable, Dict, List, Optional, Sequence, Union, cast
 
 import numpy as np
 import torch
-from einops import repeat
-from sklearn.base import BaseEstimator
+from einops import rearrange, reduce, repeat
+from sklearn.base import BaseEstimator, clone
 from sklearn.ensemble import RandomForestClassifier, RandomForestRegressor
 from sklearn.linear_model import LinearRegression, LogisticRegression
 from sklearn.multioutput import MultiOutputClassifier
-from torch.utils.data import DataLoader, TensorDataset
+from torch.optim import Optimizer
+from torch.utils.data import DataLoader, Dataset, default_collate
+from tqdm import tqdm
 
+from ..dataops import S1_S2_ERA5_SRTM, DynamicWorld2020_2021
 from ..model import FineTuningModel, Seq2Seq
 from ..utils import DEFAULT_SEED, device
 from .knn import KNNat5, KNNat20, KNNat100
 
+logger = logging.getLogger("__main__")
 
-class EvalDataset:
+
+@dataclass
+class Hyperparams:
+    lr: float = 3e-4
+    max_epochs: int = 20
+    batch_size: int = 4096
+    patience: int = 3
+    num_workers: int = 2
+
+
+class EvalTask(ABC):
     name: str
     num_outputs: int
     regression: bool
     multilabel: bool
 
+    def __init__(self, seed: int = DEFAULT_SEED):
+        self.seed = seed
```

Comment:
I don't really like that this class has a `seed: int` argument and its subclass EvalTaskWithAggregatedOutput has a `seeds: List[int]` argument, so you basically have to use `isinstance(EvalTaskWithAggregatedOutput,...)` to know what field to use. I think it would be nicer to have this class use a field `seeds: List[int]` too and include a check this its length equals 1 (or something along these lines)

Reply:
updated!